scripts/pre_commit_linter.py

```diff
@@ -1357,6 +1357,62 @@ def _check_html_indent(all_files, debug=False):
     return summary_messages
 
 
+def _check_for_copyright_notice(all_files):
+    """This function checks whether copyright notice is
+    present at the beginning of files.
+    """
+
+    js_files_to_exclude = [
+        'generatedDefaultData.js', 'generatedParser.js', 'protractor.conf.js',
+        'karma.conf.js', 'mathjaxConfig.js', 'ExpressionParserService.js',
+        'constants.js', 'rich_text_components_definitions.js']
+    js_files_to_check = [
+        filename for filename in all_files if filename.endswith('.js') and (
+            not filename.endswith(tuple(js_files_to_exclude)))]
+    py_files_to_check = [
+        filename for filename in all_files if filename.endswith('.py') and (
+            not filename.endswith('__init__.py'))]
+    sh_files_to_check = [
+        filename for filename in all_files if filename.endswith('.sh')]
+    all_files_to_check = (
+        js_files_to_check + py_files_to_check + sh_files_to_check)
+    statement_to_check = 'All Rights Reserved.'
+
+    failed = False
+    summary_messages = []
+
+    for filename in all_files_to_check:
+        has_copyright_notice = False
+        with open(filename, 'r') as f:
+            for line in f:
+                if 'Copyright' in line and statement_to_check in line:
```

Comment:
This check should be a regex check where we ensure that it looks like:

`Copyright %d%d%d%d The Oppia Authors. All Rights Reserved.`

The Apache 2.0 license specifies a format that we have to follow. Some files are legacy and may break this check, you might want @seanlip's opinion on whether we want to change those headers so it looks like the one above.

Reply:
There are a couple of files with the following:
`Copyright 2012 Google Inc. All Rights Reserved.`

The above-mentioned check will fail on these.

Reply:
Yup, which is why you'll need Sean's opinion for those :)

Reply:
You might have to whitelist patterns, but let's see what he says.

Reply:
The check I have used works for all cases. We do not need to whitelist patterns if we use a regexp:
`Copyright \d+ .[^\.]+. All Rights Reserved.`

Reply:
Can you send me links to all the files which deviate from the norm? That might just be a historical error.

Reply:
All these files do not have copyright notices:
extensions/interactions/LogicProof/static/js/generatedDefaultData.js
extensions/interactions/LogicProof/static/js/generatedParser.js 
core/tests/protractor.conf.js 
core/tests/karma.conf.js
core/templates/dev/head/mathjaxConfig.js
core/templates/dev/head/expressions/ExpressionParserService.js
assets/constants.js
assets/rich_text_components_definitions.js

I am not sure if we require notices for them too.

All these files have copyright notices for Google Inc.:
extensions/objects/templates/MusicPhraseEditor.js
extensions/objects/templates/NonnegativeIntEditor.js
extensions/objects/templates/SetOfHtmlStringEditor.js

Reply:
OK, thanks!

- The copyright notices for the last three files should be updated.
- Copyright notices need not be included for generated files, since regeneration will just get rid of those. This covers: generatedDefaultData, generatedParser, ExpressionParserService. You can have a GENERATED_FILE_PATHS constant for these (or a comment above the list of exclusions to explain why they're being excluded).
- The remaining files are basically configs. OK to put copyright notices for them, if it simplifies linting.

Reply:
Okay, I will update the notices for the last three files. For the remaining, I will create a GENERATED_FILE_PATHS and for the configs, I will create another CONFIG_FILE_PATHS. We can then exclude both of these paths. Does this sound good?

Reply:
Yep, thanks!